test/suites/dev/test-precompile/test-precompile-multiple-locks.ts

```diff
@@ -0,0 +1,57 @@
+import "@moonbeam-network/api-augment";
+import { beforeAll, describeSuite, expect } from "@moonwall/cli";
+import {
+  GLMR,
+  MIN_GLMR_DELEGATOR,
+  alith,
+  checkBalance,
+  createRawTransfer,
+  generateKeyringPair,
+} from "@moonwall/util";
+import { createProposal } from "../../../helpers/voting.js";
+
+describeSuite({
+  id: "D2580",
+  title: "Precompiles - Locks - Voting and staking locks are not mutually exclusive",
+  foundationMethods: "dev",
+  testCases: ({ context, it }) => {
+    let proposalIndex: number;
+    const randomAccount = generateKeyringPair();
+    const randomAddress = randomAccount.address as `0x${string}`;
+    beforeAll(async function () {
+      await context.createBlock(
+        createRawTransfer(context, randomAddress, MIN_GLMR_DELEGATOR + GLMR)
+      );
+      proposalIndex = await createProposal(context);
+    });
+
+    it({
+      id: "T01",
+      title: "should be able to vote (conviction) and stake",
+      test: async function () {
+        // Vote yes on proposal with 1 GLMR
+        await context.createBlock(
+          context
+            .polkadotJs()
+            .tx.convictionVoting.vote(proposalIndex, { Standard: { Vote: 0, Amount: GLMR } })
```

Comment:
We don't get any typecheck with `convictionVoting.vote`, but that vote is not valid. Running the test using Moonwall's `run` mode and connecting to the node through PolkadotJS apps we can see what was executed:
![image](https://github.com/moonbeam-foundation/moonbeam/assets/52399794/d13b199a-a183-461d-b9dd-ca2148d1585b)

It doesn't help that it still accepts the wrong object and doesn't complain at all, it just "defaults" to a null 0 balance vote, which doesn't lock any funds.

```suggestion
            .tx.convictionVoting.vote(proposalIndex, {
              Standard: { vote: { aye: true, conviction: "Locked5x" }, balance: GLMR },
            }
```